src/sankeydiagram.ts

```diff
@@ -420,20 +475,46 @@ export class SankeyDiagram implements IVisual {
             dataView.categorical.categories[SankeyDiagram.DestinationCategoryIndex].source.displayName,
             dataView.categorical.values ? dataView.categorical.values[SankeyDiagram.FirstValueIndex].source.displayName : null
         );
-
         let cycles: SankeyDiagramCycleDictionary = this.checkCycles(nodes);
-
         if (settings.cyclesLinks.drawCycles === CyclesDrawType.Duplicate) {
             links = this.processCyclesForwardLinks(cycles, nodes, links, settings);
         }
-
-        let sankeyDiagramDataView = {
+        const savedNodes = new Set<SankeyDiagramNode>();
+        settings._linkPositions.forEach(l => { savedNodes.add(l.source); savedNodes.add(l.destination); });
+        let savedNodesArray: SankeyDiagramNode[] = [];
+        savedNodes.forEach(node => savedNodesArray.push(node));
+        let sankeyDiagramDataView: any = {
             nodes,
             links,
             settings,
             columns: []
         };
-
+        settings.filterDetected = savedNodesArray.length !== nodes.length;
+        if (settings.filterDetected) {
+            settings._linkPositions.forEach(l => l.display = true);
+            if (settings.general.pinNodes) {
+                settings._linkPositions.forEach(link => {
+                    link.identity = this.visualHost.createSelectionIdBuilder().createSelectionId();
+                    link.display = false;
+                });
+                links.forEach(link => {
+                    let foundLink: SankeyDiagramLink = settings._linkPositions.find(slink => slink.source.label.name === link.source.label.name && slink.destination.label.name === link.destination.label.name);
+                    if (foundLink) { this.saveLinkProperties(link, foundLink); }
+                });
+                sankeyDiagramDataView.nodes = savedNodesArray;
+                sankeyDiagramDataView.links = settings._linkPositions;
+            }
+        }
+        else {
+            links.forEach(link => {
+                let foundLink: SankeyDiagramLink = settings._linkPositions.find(slink => slink.source.label.name === link.source.label.name && slink.destination.label.name === link.destination.label.name);
+                if (foundLink) {
+                    this.saveLinkProperties(link, foundLink);
+                }
+            });
+            sankeyDiagramDataView.nodes = savedNodesArray;
+            sankeyDiagramDataView.links = settings._linkPositions;
```

Comment:
Are **links** and **_linkPositions** variables the same? If so, why do you use different names?

Reply:
Not the same. _linkPositions always keeps all link data of an application, links keep data that is required for rendering. If filter was applied _linkPositions will still have all the data parsed from appState, but links will only contain a subset - the links that should be rendered.